Avoid updating the XAPI database from slaves. Master should be enough,
and the command can be run on several hosts at the same time when updating
the hosts via yum, thus risking to break the XAPI database.

--- a/create-guest-templates	2019-09-03 15:40:29.196315228 +0000
+++ b/create-guest-templates	2019-09-03 15:41:07.671640196 +0000
@@ -9,6 +9,14 @@
         session = XenAPI.xapi_local()
         session.xenapi.login_with_password('', '', '', 'create-template')
 
+        xapiSession = session.xenapi.session.get_record(session._session)
+        pool = session.xenapi.pool.get_all()[0]
+        masterRef = session.xenapi.pool.get_master(pool)
+
+        if masterRef != xapiSession['this_host']:
+            print "This node is not pool master, skip update"
+            exit(0)
+
         loader = loader.Loader(session)
         loader.find_config_files("/usr/share/xapi/vm-templates",
                                  "/etc/xapi.d/vm-templates")
